packages/Uploader/src/helpers.js

```diff
@@ -0,0 +1,122 @@
+import uuidv4 from "uuid/v4";
+import superagent from "superagent";
+import types from "./types";
+
+export function fileSizeUnitsToHumanReadableFormat(size) {
+  const aMultiples = ["KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"];
+  let nMultiple = 0;
+  let sOutput = "";
+  // this calculate Kibebytes, Mebibyes, etc.
+  // But are being label as KB and MB for easier understanding.
+  for (let nApprox = size / 1024; nApprox > 1; nApprox /= 1024, nMultiple++) {
+    // fixed to 2 decimals but we could add more if need it
+    sOutput = `${nApprox.toFixed(2)} ${aMultiples[nMultiple]}`;
+  }
+
+  return sOutput;
+}
+
+function getExtension({ file }) {
+  const filename = file.name;
+  return filename.substr((~-filename.lastIndexOf(".") >>> 0) + 2); // eslint-disable-line
+}
+
+function isValidFileType({ file, okFileTypes }) {
+  const validMimeType = (validFiles, type) => {
+    const mimetype = type.split("/");
+    if (mimetype.length > 0) {
+      return validFiles.indexOf(mimetype[0].toUpperCase()) > -1;
+    }
+    return false;
+  };
+
+  if (okFileTypes.length === 1 && okFileTypes[0] === "*/*") {
+    return true;
+  }
+  // copy/pasta from acl-ui-3
+  // http://stackoverflow.com/a/1203361/196038
+  const validFiles = okFileTypes.join("").toUpperCase();
+  const extension = getExtension({ file });
+
+  return validFiles.indexOf(extension.toUpperCase()) > -1 || validMimeType(validFiles, file.type);
+}
+
+function isValidFile({ file, maxFileSize, okFileTypes }) {
+  const validation = {
+    isServerValid: true, // this will be true unless failed when uploading the file
+    isSizeValid: file.size <= maxFileSize,
+    isTypeValid: isValidFileType({ file, okFileTypes }),
+  };
+
+  validation.isValid = [validation.isSizeValid, validation.isTypeValid].every(rule => rule);
+
+  return validation;
+}
+
+function createFilesDataStructure({ files, maxFileSize, okFileTypes, endpoint }) {
+  return [...files].map(file => {
```

Comment:
```suggestion
  return files.map(file => {
```
Don't need a new array since you're only mapping over